File: Backend/Library/BJB.Dashboard.Library/Configuration/ConfigurationBuilders.cs
using System;
using Microsoft.Extensions.Configuration;

namespace BJB.Dashboard.Library.Configuration;

public class ConfigurationBuilders
{
    public static void BuildConfiguration(IConfiguration configuration)
    {
        ConfigurationDictionary.ReadConfig("SqlServerDbContext", configuration.GetSection("ConnectionStrings:SqlServerDbContext").Value ?? throw new InvalidOperationException("SqlServerDbContext is not configured."));
        ConfigurationDictionary.ReadConfig("ApplicationDatabaseEngine", configuration.GetSection("ApplicationConfig:DBEngine").Value ?? throw new InvalidOperationException("DBEngine is not configured."));
    }
}

File: Frontend/src/app/configuration/configuration-builders.ts
import { ConfigurationDictionary } from './configuration-dictionary';

export interface FrontendSettings {
  api?: {
    baseUrl?: string;
  };
}

export class ConfigurationBuilders {
  static buildConfiguration(settings: FrontendSettings): void {
    const apiBaseUrl = settings.api?.baseUrl;

    if (!apiBaseUrl) {
      throw new Error('Api:BaseUrl is not configured.');
    }

    ConfigurationDictionary.ReadConfig('ApiBaseUrl', apiBaseUrl);
  }
}
